### DISCUSSION TITLE
[Android] Take over support of Kotlin Synthetics

### DISCUSSION THREAD

**Discussion by mkalinowski2** [MobileNativeFoundation/discussions#45] [Proposals]
Sadly, Kotlin Synthetics lost Jetbrains support because Jetpack ViewBinding.

I don't think that new (second) library is enough reason to justify abandon Synthetics library which works really nice and is so different from ViewBinding.

Personally I think that both libraries have some pluses and minuses. But I don't think that advantages of ViewBinding are worth to rewrite ready project which was written in Synthetics.

I wish if it could be still supported and improved, so developers would have a choice if they want to use Synthetics or ViewBinding.

Synthetics library is opensource and it's possible to take support: https://github.com/JetBrains/kotlin/tree/master/plugins/android-extensions

**Comment by Plastix** [comment]
I know this isn't the answer you want to hear but my team decided to just bite the bullet and migrate away from Kotlin Synthetics to View Binding last year. We've had bugs because it is easy to import the wrong synthetic binding. In some cases such as dynamic layouts with A/B experiments, you basically can't use synthetics at all.

We're very happy with View Binding, although it does have its own set of gotchas (e.g., choosing the right root layout to bind and nested included layouts). We're in the process of migrating our legacy Butterknife code over to View Binding now.
